packages/editor/src/editor/sidebar/fields/datatable-columns/ColumnsCheckboxField.tsx

```diff
@@ -20,7 +20,12 @@ export const ColumnsCheckboxField = ({ onChange }: ColumnsCheckboxFieldProps) =>
         config: DataTableColumnComponent.create({
           label: column.config.header,
           value: column.config.value,
-          defaultProps: { filterable: column.config.filterable, sortable: column.config.sortable }
+          defaultProps: {
+            filterable: column.config.filterable,
+            sortable: column.config.sortable,
+            visible: column.config.visible,
+            header: column.config.header
+          }
```

Comment:
still do not understand why you have to add the default props here. They should already be present on the creation method. (defaultDataTableColumnProps)
![image](https://github.com/user-attachments/assets/30f4aa12-7aa5-4091-bfc4-666f55a96802)

Reply:
Because in this case, I intentionally want to modify the defaultProps. When I select a column via checkbox and make changes to it (e.g., header or sortable), and then deselect and reselect it, the regular defaultProps would reset sortable/filterable to false and visible to an empty string (""). This is not the behavior I want, which is why I intentionally store the column in a local state and set the properties based on that last state. I hope this explanation is clear, but if not, we can discuss it further in person.